tests/cpp/tree/hist/test_evaluate_splits.cc

```diff
@@ -39,7 +36,7 @@ template <typename GradientSumT> void TestEvaluateSplits() {
   std::iota(row_indices.begin(), row_indices.end(), 0);
   row_set_collection.Init();
 
-  auto hist_builder = GHistBuilder<GradientSumT>(n_threads, gmat.cut.Ptrs().back());
+  auto hist_builder = GHistBuilder<GradientSumT>(omp_get_max_threads(), gmat.cut.Ptrs().back());
```

Comment:
Shouldn't we call function `OmpGetThreadLimit()` to query `OMP_THREAD_LIMIT`?

Reply:
It's a c++ unittest, I think we don't have to worry about that too much.  We on the other hand, we need for thorough  integration tests for that env, probably in another PR.